web/modules/custom/hello_user/src/CurrentUser.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace Drupal\hello_user;
+
+use Drupal\Core\Session\AccountProxyInterface;
+
+/**
+ * Provides a service to get current user's display name.
+ */
+class CurrentUser {
+
+  /**
+   * The current user.
+   * 
+   * @var \Drupal\Core\Session\AccountProxyInterface
+   */
+  protected $currentUser;
+
+  /**
+   * Constructs a new CurrentUser object.
+   * 
+   * @param \Drupal\Core\Session\AccountProxyInterface $currentUser
+   *   The current user service.
+   */
+  public function __construct(AccountProxyInterface $currentUser) {
+    $this->currentUser = $currentUser;
+  }
+
+  /**
+   * Gets the current user's display name.
+   * 
+   * @return string
+   *   The display name.
+   */
+  public function getUserDisplayName() {
+    return $this->currentUser->getDisplayName();
```

Comment:
Why create a new function for funtionality which already exists?

Reply:
Earlier has added code like:
`\Drupal::currentUser()->getAccountName();`
You added comment that it was not the proper way and check services and dependencies. Should it be done some other way?

Reply:
Check your first pull request.
There you have done it correctly :)

Reply:
Actually This file is same as the first pull request. I have not made any changes to this file ;)

Reply:
Yes, I see that now.
I think you are not using this service anymore.
So please remove the service and its definition.

Reply:
If I remove the service and its definition, then the module hello_user will not work as it depends on the service. It is actually for the task FT2023-339. Should I remove the entire module hello_user then?

Reply:
Please remove the service only not the module.
What you have done in this service is something which Drupal core service already does, so this is not adding any value to the existing code.
If you want to show that you have created a service, then please do something more meaningful in that.

Reply:
Okay removed the service.